src/main/java/dev/morling/onebrc/CalculateAverage_felix19350.java

```diff
@@ -74,62 +76,73 @@ public void mergeResult(ResultRow value) {
     }
 
   private record AverageAggregatorTask(MemorySegment memSegment) {
+    private static final int[] INT_PARSE_FACTORS = new int[]{1, 10, 100};
+    private static final int HASH_FACTOR = 607; // Mersenne prime
+
 
     public static Stream<AverageAggregatorTask> createStreamOf(List<MemorySegment> memorySegments) {
       return memorySegments.stream().map(AverageAggregatorTask::new);
     }
 
     public Map<String, ResultRow> processChunk() {
-      final var result = new TreeMap<String, ResultRow>();
+      final var measurements = new HashMap<Integer, ResultRow>(EXPECTED_MAX_NUM_CITIES);
+      final var cityNames = new HashMap<Integer, String>(EXPECTED_MAX_NUM_CITIES);
```

Comment:
What's used as the key here? How does it prevent collisions between entries?

Reply:
Hmm good point, it doesn't really. Let me see what I can do in terms of proper perfect hashing :)

Reply:
@gunnarmorling addressed your comment above (check the equals method on the `CityRef` class) and squeezed a bit more performance out of this while (hopefully) maintaining a relatively readable and idiomatic solution.